proxy/httpproxy/handler.go

```diff
@@ -0,0 +1,70 @@
+// Copyright 2019 PingCAP, Inc.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package httpproxy
+
+import (
+	"net/http"
+
+	"github.com/gorilla/mux"
+	"github.com/tikv/client-go/proxy"
+)
+
+// NewHTTPProxyHandler creates an http.Handler that serves as a TiKV client proxy.
+func NewHTTPProxyHandler() http.Handler {
+	router := mux.NewRouter()
+	rawkv := rawkvHandler{p: proxy.NewRaw()}
+
+	router.HandleFunc("/rawkv/client/new", rawkv.handlerFunc(rawkv.New))
```

Comment:
em, I prefer using raw and txn directly

Reply:
ok

Reply:
em, `/txn/txn/{id}/get` will be strange then, `txnkv` seems to still be the better option. @siddontang

Reply:
IMO, why not providing a gRPC support instead of Restful. For Restful, we have to care how to manage transaction ID, if the client forgets to rollback the transaction, we have to rollback it soon.

But in gRPC, we can manage the transaction in a session, it is more simple.

Reply:
Not quite familiar with _grpc session_ you mentioned. But I think HTTP is a better idea than gRPC because it is supported by almost all modern languages. In general the developer of the client can introduce it with less effort.